coronavirus/coronavirus.py

```diff
@@ -323,7 +323,33 @@ def fetch_data_germany(include_last_day=True):
     return cleaned
 
 
-def germany_get_region(state=None, landkreis=None):
+def pad_cumulative_series_to_yesterday(series):
+    """Given a time series with date as index and cumulative cases/deaths as values:
+
+    - if the last date in the index is older than yesterday, then
+    - add that date
+    - resample the series with a daily interval, using padding with last known value
+    - and return.
+
+    Required for Robert Koch Data, where only a new data point is provided if
+    the numbers change, but the plotting algorithms need to know that there is
+    no change. Without this padding, the data set looks old as the last plotted
+    data point is the last one for which data is provided.
+    """
+    now = datetime.datetime.now()
+    today = now.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
Minor comment:

`datetime.datetime.now()` would return the current time in the users timezone, if somebody was calling this from a different timezone ahead of Germany it would end up padding the data up to their current day even though if it might not be that day yet in Germany.

`pytz` is included at a standard library for dealing with timezones, from a glance I think that this would (probably?) be fixed by doing something like:

```
import pytz

rki_tz = pytz.timezone('Europe/Berlin')
rki_now = datetime.now(rki_tz)
```

Then `rki_today = rki_now.replace(hour=0, minute=0, second=0, microsecond=0)` would be 'today' in Germany.

Realistically, this won't make much of a difference since it would only happen rarely, and when it does happen it doesn't really affect the results.

So LGTM :+1:

Reply:
Well spotted. I put this in. Worked as you suggested, just had to remove the timezone info subsequently to allow comparison (">") between other datetime object we use.